open-banking-accelerator/components/consent-management/com.wso2.openbanking.accelerator.consent.extensions/src/main/java/com/wso2/openbanking/accelerator/consent/extensions/authorize/utils/ConsentRetrievalUtil.java

```diff
@@ -256,4 +272,429 @@ public static JSONArray appendDummyAccountID() {
+                consentDataJSON = populateAccountData(data, consentDataJSON);
+                break;
+            case ConsentExtensionConstants.PAYMENTS:
+               consentDataJSON = populateSinglePaymentData(data, consentDataJSON);
+                // Adding Debtor Account
+                ConsentRetrievalUtil.populateDebtorAccount(data, consentDataJSON);
+                // Adding Creditor Account
+                ConsentRetrievalUtil.populateCreditorAccount(data, consentDataJSON);
+                break;
+            case ConsentExtensionConstants.FUNDSCONFIRMATIONS:
+                consentDataJSON = populateCofData(data, consentDataJSON);
+                break;
+            case ConsentExtensionConstants.VRP:
+                consentDataJSON = populateVRData(controlParameters, consentDataJSON);
+                break;
+            default:
+                break;
+        }
+        return consentDataJSON;
+
+    }
+
+    /**
+     * Method defined to retrieve the Payment consent related data in the authorization flow to send them to the
+     * consent page to get PSU consent.
+     *
+     * @param consentResource Consent Resource parameter containing consent related information retrieved from database
+     */
+    public static JSONArray getConsentRetrievalDataSet(ConsentResource consentResource) {
+
+        try {
+            String receiptString = consentResource.getReceipt();
+            Object receiptJSON = new JSONParser(JSONParser.MODE_PERMISSIVE).parse(receiptString);
+
+            //Checking whether the request body is in JSON format
+            if (!(receiptJSON instanceof JSONObject)) {
+                log.error(ErrorConstants.NOT_JSON_OBJECT_ERROR);
+                throw new ConsentException(ResponseStatus.INTERNAL_SERVER_ERROR, ErrorConstants.NOT_JSON_OBJECT_ERROR);
+            }
+
+            //Checking whether the consent status is valid
+            if (!consentResource.getCurrentStatus().equals(ConsentExtensionConstants.AWAITING_AUTH_STATUS)) {
+                log.error(ErrorConstants.STATE_INVALID_ERROR);
+                //Currently throwing error as 400 response. Developer also have the option of appending a field IS_ERROR
+                // to the jsonObject and showing it to the user in the webapp. If so, the IS_ERROR have to be checked in
+                // any later steps.
+                throw new ConsentException(ResponseStatus.BAD_REQUEST, ErrorConstants.STATE_INVALID_ERROR);
+            }
+        } catch (ParseException e) {
+            throw new ConsentException(ResponseStatus.INTERNAL_SERVER_ERROR, ErrorConstants.CONSENT_RETRIEVAL_ERROR);
+        }
+        return null;
+
+    }
+
+    /**
+     * Populate Domestic and international Payment Details.
+     *
+     * @param data       data request from the request
+     * @param consentDataJSON   Consent information
+     */
+    private static JSONArray populateSinglePaymentData(JSONObject data, JSONArray consentDataJSON) {
```

Comment:
Shall we remove single part and just say populatePaymentData

Reply:
Fixed in [a6d2b0b]